packages/lodestar/src/network/gossip/validator.ts

```diff
@@ -42,90 +44,175 @@ export class GossipMessageValidator implements IGossipMessageValidator {
+        e.code === AttestationErrorCode.ERR_KNOWN_BAD_BLOCK ||
+        e.code === AttestationErrorCode.ERR_FINALIZED_CHECKPOINT_NOT_AN_ANCESTOR_OF_ROOT ||
+        e.code === AttestationErrorCode.ERR_TARGET_BLOCK_NOT_AN_ANCESTOR_OF_LMD_BLOCK ||
+        e.code === AttestationErrorCode.ERR_INVALID_INDEXED_ATTESTATION
+      ) {
+        this.logger.warn("Rejecting gossip attestation", e.toObject());
+        return ExtendedValidatorResult.reject;
+      } else if (
+        e.code === AttestationErrorCode.ERR_SLOT_OUT_OF_RANGE ||
+        e.code === AttestationErrorCode.ERR_ATTESTATION_ALREADY_KNOWN ||
+        e.code === AttestationErrorCode.ERR_UNKNOWN_HEAD_BLOCK ||
+        e.code === AttestationErrorCode.ERR_MISSING_ATTESTATION_PRESTATE
+      ) {
+        this.logger.warn("Ignoring gossip attestation", e.toObject());
+        return ExtendedValidatorResult.ignore;
+      }
     }
+    return ExtendedValidatorResult.accept;
   };
 
   public isValidIncomingAggregateAndProof = async (
     signedAggregationAndProof: SignedAggregateAndProof
   ): Promise<ExtendedValidatorResult> => {
     try {
-      return await validateGossipAggregateAndProof(
+      const attestationJob = {
+        attestation: signedAggregationAndProof.message.aggregate,
+        validSignature: false,
+      } as IAttestationJob;
+      await validateGossipAggregateAndProof(
         this.config,
         this.chain,
         this.db,
         this.logger,
-        signedAggregationAndProof
+        signedAggregationAndProof,
+        attestationJob
       );
+      await this.updateAttestationSeenCaches(signedAggregationAndProof.message);
     } catch (e) {
       this.logger.error("Error while validating gossip aggregate and proof", e);
-      return ExtendedValidatorResult.ignore;
+      if (
+        e.code === AttestationErrorCode.ERR_WRONG_NUMBER_OF_AGGREGATION_BITS ||
+        e.code === AttestationErrorCode.ERR_KNOWN_BAD_BLOCK ||
+        e.code === AttestationErrorCode.ERR_AGGREGATOR_NOT_IN_COMMITTEE ||
+        e.code === AttestationErrorCode.ERR_INVALID_SELECTION_PROOF ||
+        e.code === AttestationErrorCode.ERR_INVALID_SIGNATURE ||
+        e.code === AttestationErrorCode.ERR_INVALID_INDEXED_ATTESTATION ||
+        e.code === AttestationErrorCode.ERR_INVALID_AGGREGATOR
+      ) {
+        this.logger.warn("Rejecting gossip aggregate & Proof", e.toObject());
+        return ExtendedValidatorResult.reject;
+      } else if (
+        e.code === AttestationErrorCode.ERR_SLOT_OUT_OF_RANGE ||
+        e.code === AttestationErrorCode.ERR_AGGREGATE_ALREADY_KNOWN ||
+        e.code === AttestationErrorCode.ERR_MISSING_ATTESTATION_PRESTATE
+      ) {
+        this.logger.warn("Ignoring gossip aggregate & Proof", e.toObject());
+        return ExtendedValidatorResult.ignore;
+      }
+      await this.updateAttestationSeenCaches(signedAggregationAndProof.message);
```

Comment:
- you have same line in 139, if you move it outside of try/catch then you can call it just once.
- this line here could be dangerous, we had unknown error and you stored it into our aggregate pool which means it will end up in next block we try to produce and we would fail any block production onwards

Reply:
ok, i remove the aggregate pool addition and kept the seenCache addition and moved it outside the try/catch